src/components/Checkbox/Checkbox.style.ts

```diff
@@ -0,0 +1,62 @@
+import styled from 'styled-components';
+import { ThemeColor, ThemeElementAppearance, ThemeElementSize } from '../../types.ts';
+import { readableColor, shade } from 'polished';
+
+type StyledButtonProps = {
+  $color: ThemeColor;
+  $appearance: ThemeElementAppearance;
+  $size: ThemeElementSize;
+};
+
+export const StyledCheckbox = styled.button<StyledButtonProps>`

```

Comment:
Semantically, a checkbox should be an [input checkbox](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox), not a button. Guessing you did it this way because options for styling actual checkboxes are pretty limited; you can get around this with some trickery using pseudo-elements. Here's a resource with examples: https://css-tricks.com/the-checkbox-hack/